services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -370,9 +370,9 @@ async def list_projects(  # pylint: disable=too-many-arguments
                 )
 
             if order_by.direction == OrderDirection.ASC:
-                query = query.order_by(sa.asc(order_by.field))
+                query = query.order_by(sa.asc(getattr(projects.c, order_by.field)))
```

Comment:
Please validate and handle the error here properly. 
If  `getattr(projects.c, order_by.field)` raises (NOTE that `OrderBy.field` is just constraint to a `str`) what do you want to do?

Options:
- error and respond with 40X since the field is invalid?
- default to a given sort


Another option is that you enforce already some "valid/allowed" fields at the `OrderBy.field` level. Perhaps inheriting as `ProjectOrderBy` and selecting some valid columns of `projects` table?

Reply:
validation is already done here `ProjectListWithJsonStrParams`

Reply:
OK. it was difficult to deduce from this function's signature. In that case, please add a simple `assert` to document that gurantee in your code


```python
    async def list_projects(  # pylint: disable=too-many-arguments
        self,
        user_id: PositiveInt,
        *,
        product_name: str,
        filter_by_project_type: ProjectType | None = None,
        filter_by_services: list[dict] | None = None,
        only_published: bool | None = False,
        include_hidden: bool | None = False,
        offset: int | None = 0,
        limit: int | None = None,
        search: str | None = None,
        order_by: OrderBy = OrderBy(
            field="last_change_date", direction=OrderDirection.DESC
        ),
    ) -> tuple[list[dict[str, Any]], list[ProjectType], int]:
    
,    assert order_by.field in projects.column, "Guaranteed by ProjectListWithJsonStrParams" # nosec
    
    ```

Reply:
Ok, I like that, will do, thanks.